#!/usr/bin/env python
PACKAGE = "iris_cobot"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("leaf_size"         ,double_t,   0, "Voxel down sample leaf size",      0.03, 0.005, 1)
gen.add("radius_search"     ,double_t,   0, "Search readius for KDTree",        0.1, 0.005, 1)
gen.add("cluster_tolerance" ,double_t,   0, "Clustering algorithm tolerance",   0.1, 0.005, 20)

gen.add("min_cluster_size", int_t,      0, "Min cluster size (Points / X)",     1, 1, 100)
gen.add("max_cluster_size", int_t,      0, "Max cluster size (Points / X)",     90, 1, 100)
gen.add("normal_diff",      double_t,   0, "Minimum normal difference",         0.9999, 0.001, 1)
gen.add("squared_dist",     double_t,   0, "Minimum squared distance",          0.001, 0.00000001, 10)

exit(gen.generate(PACKAGE, "iris_cobot", "Clustering"))